pages/en/lb4/Decorators.md

```diff
@@ -0,0 +1,379 @@
+    async whoAmI() : Promise<string> {
+      return this.user.id;
+    }
+  }
+  ```
+
+## Repository Decorators
+
+  As a Domain-driven design concept, 
+  repository means a mediate between your domain object and data mapping layers
+  using a collection-like interface for accessing domain objects. 
+  In framework LoopBack, a domain object is usually a TypeScript/JavaScript Class instance, 
+  and a typical example of data mappting layer module could be a database's node.js driver.
+  
+  LoopBack repository encapsulates your TypeScript/JavaScript Class instance, 
+  and its methods that communicate with your database.
+  It is an interface to implement data persistency. 
+
+  Repository decorators are used for defining models(domain objects) for use with your chosen datasources, 
+  and the navigation strategies among models.
+
+  If you are not familiar with repository related concepts like Model, Entity, Datasource,  
+  please see LoopBack concept [Repositories](#Repositories.htm) to learn more. 
+  
+### Model Decorators
+
+  Model is a Class that LoopBack builds for you to organize the data that 
+  share same configurations and properties.
+  You can use model decorators to define a model and its properties.
+
+#### Model Decorator
+
+  Syntax: `@model(definition?: ModelDefinitionSyntax)`
+
+  Model decorator is a Class decorator.
+  In LoopBack 4, we inherit the model definition format from LoopBack 3,
+  you can find it in [Model definition JSON file](https://loopback.io/doc/en/lb3/Model-definition-JSON-file.html).
+  For usage examples, see [Define Models](Repositories.html#define-models)
+
+  *Please note we will elaborate more about model and model definition in #Model.htm,*
+  *and replace the link above with LB4 link*
+
+  By using a model decorator, you can define a model as your repository's metadata,
+  and resolve your repository with Legacy Juggler that defautly comes with a set of CRUD operations bound to your model.
+  A use case can be found in section [Repository decorator](#repository-decorator)
+
+  Or you also have the flexibility to define a model without being resolved with Legacy Juggler.
+  This allows you to implement database access functions by yourself. 
+
+   ```ts
+  // Missing example here
+  // Will be provided in Model.htm
+  // refer to [example code](https://github.com/strongloop/loopback-next-example/blob/master/services/account-without-juggler/repositories/account/models/Account.ts)
+  ```
+
+#### Property Decorator
+
+  Syntax: `@property(definition: PropertyDefinition)`
+
+  It is a Class property decorator, defines the metadata for a Model property.
+  The format of property definitions can be found in [Property definitions](https://loopback.io/doc/en/lb2/Model-definition-JSON-file.html#properties) 
+
+  For usage examples, see [Define Models](Repositories.html#define-models)
+
+### Relation Decorators
+
+  Relation represents how you navigate from one model to other models.
```

Comment:
The relation decorator defines the nature of a relationship between two models.

Reply:
actually I prefer to explain the relation as a "navigation" concept, otherwise people will be confused what does "relationship" mean, if they are not familiar with ORM relation.

I think it's a good idea to keep them both.

Reply:
I disagree. "Navigate" can mean all sorts of confusing things.
"What do they mean by navigate? I need to navigate between my Models? Does that mean one of them calls the other?"

It's not really the correct word for what's happening here. It's not a traversal of anything, it's the established relationship between two data types. Whether or not that means one owns the other, has many of them, must be created when the other is created (and so on) doesn't have to be defined here.

One might "navigate" an existing relationship, but this is _how you make that relationship exist_.

Part of the problem is that you can't really be more specific until we have code examples that represent what's going on (whenever we start getting close to Juggler GA).

Reply:
I think in LB3, relation means you can access/modify/delete other models' data by a model's apis, which fits `navigation` more than `relationship`, but if that's not what we are going to build in LB-next then I am good with the more general description "relationship"